tests/rust_reflection_test/src/lib.rs

```diff
@@ -0,0 +1,1324 @@
+
+    let res = unsafe {
+        set_string(
+            &mut buffer,
+            &string_field,
+            "s",
+            &root_as_schema(schema.as_slice()).unwrap(),
+        )
+    }
+    .unwrap();
+
+    assert_eq!(res, true);
+    let updated_table = unsafe { get_any_root(&buffer) };
+    let updated_value = unsafe { get_field_string(&updated_table, &string_field) };
+    assert_eq!(updated_value, Some("s"));
+}
+
+#[test]
+fn test_buffer_set_string_diff_type_fails() {
+    let mut buffer = create_test_buffer();
+    let schema = load_file_as_buffer("../monster_test.bfbs");
+    let f32_field = get_schema_field(&schema, "testf");
+
+    let res = unsafe {
+        set_string(
+            &mut buffer,
+            &f32_field,
+            "any",
+            &root_as_schema(schema.as_slice()).unwrap(),
+        )
+    }
+    .unwrap();
+
+    assert_eq!(res, false);
+}
+
+#[test]
+fn test_verify_buffer_default_options_succeeds() {
+    let buffer = create_test_buffer();
+    let schema_buffer = load_file_as_buffer("../monster_test.bfbs");
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+
+    let res = verify(&buffer, &schema);
+
+    assert!(res.is_ok());
+}
+
+#[test]
+fn test_verify_buffer_limit_max_depth_fails() {
+    let buffer = create_test_buffer();
+    let schema_buffer = load_file_as_buffer("../monster_test.bfbs");
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+    let verify_options = VerifierOptions {
+        max_depth: 1,
+        ..Default::default()
+    };
+
+    let res = verify_with_options(&buffer, &schema, &verify_options);
+
+    assert!(res.is_err());
+    assert_eq!(format!("{:#?}", res.err().unwrap()), "DepthLimitReached");
+}
+
+#[test]
+fn test_verify_buffer_limit_max_table_fails() {
+    let buffer = create_test_buffer();
+    let schema_buffer = load_file_as_buffer("../monster_test.bfbs");
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+    let verify_options = VerifierOptions {
+        max_tables: 1,
+        ..Default::default()
+    };
+
+    let res = verify_with_options(&buffer, &schema, &verify_options);
+
+    assert!(res.is_err());
+    assert_eq!(format!("{:#?}", res.err().unwrap()), "TooManyTables");
+}
+
+#[test]
+fn test_verify_buffer_limit_max_size_fails() {
+    let buffer = create_test_buffer();
+    let schema_buffer = load_file_as_buffer("../monster_test.bfbs");
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+    let verify_options = VerifierOptions {
+        max_apparent_size: 1 << 6,
+        ..Default::default()
+    };
+
+    let res = verify_with_options(&buffer, &schema, &verify_options);
+
+    assert!(res.is_err());
+    assert_eq!(format!("{:#?}", res.err().unwrap()), "ApparentSizeTooLarge");
+}
+
+fn load_file_as_buffer(path: &str) -> Vec<u8> {
+    let mut file = File::open(path).unwrap();
+    let mut data = Vec::new();
+    file.read_to_end(&mut data).unwrap();
+    data
+}
```

Comment:
You may like [`std::fs::read(path)`](https://doc.rust-lang.org/std/fs/fn.read.html) :smile: